unified/build_unified.sh

```diff
@@ -51,27 +51,28 @@ done
 
 UNIFIED_PKG="$(realpath -s $UNIFIED_PKG)"
 PKGS="build/$MODE/dist/tar/$PRODUCT-$VERSION-$RELEASE.$(arch).tar.gz build/$MODE/dist/tar/$PRODUCT-python3-$VERSION-$RELEASE.$(arch).tar.gz build/$MODE/dist/tar/$PRODUCT-jmx-$VERSION-$RELEASE.noarch.tar.gz build/$MODE/dist/tar/$PRODUCT-tools-$VERSION-$RELEASE.noarch.tar.gz"
+BASEDIR="build/$MODE/unified/$PRODUCT-$VERSION"
 
 rm -rf build/"$MODE"/unified/
-mkdir -p build/"$MODE"/unified/
+mkdir -p "$BASEDIR"
 for pkg in $PKGS; do
     if [ ! -e "$pkg" ]; then
         echo "$pkg not found."
         echo "please build relocatable package before building unified package."
         exit 1
     fi
     pkg="$(readlink -f $pkg)"
-    tar -C build/"$MODE"/unified/ -xpf "$pkg"
+    tar -C "$BASEDIR" -xpf "$pkg"
     dirname=$(basename "$pkg"| sed -e "s/-$VERSION_ESC-$RELEASE_ESC\.[^.]*\.tar\.gz//")
     dirname=${dirname/#$PRODUCT/scylla}
-    if [ ! -d build/"$MODE"/unified/"$dirname" ]; then
+    if [ ! -d "$BASEDIR/$dirname" ]; then
         echo "Directory $dirname not found in $pkg, the pacakge may corrupted."
         exit 1
     fi
 done
-ln -f unified/install.sh build/"$MODE"/unified/
-ln -f unified/uninstall.sh build/"$MODE"/unified/
-# relocatable package format version = 2.2
-echo "2.2" > build/"$MODE"/unified/.relocatable_package_version
+ln -f unified/install.sh "$BASEDIR"
+ln -f unified/uninstall.sh "$BASEDIR"
+# relocatable package format version = 3.0
+echo "3.0" > build/"$MODE"/unified/.relocatable_package_version
```

Comment:
I think there may be a mistake here: Before this patch, the file ".relocatable_package_version" ended up inside the output tar, but now because it's in unified/ and not in unified/$PRODUCT-$VERSION (a.k.a BASEDIR) it won't be saved in the tar. Maybe you should change this to:
```
echo "3.0" > "$BASEDIR/.relocatable_package_version"?
```

Reply:
No, it will archive build/$MODE/unified/$PRODUCT-$VERSION and build/$MODE/unified/.relocatable_package_version, so tar file contents will be like this:
```
scylla-5.2.0~dev/
scylla-5.2.0~dev/scylla/
scylla-5.2.0~dev/scylla/libexec/
scylla-5.2.0~dev/scylla/libexec/scylla
...
scylla-5.2.0~dev/install.sh
scylla-5.2.0~dev/uninstall.sh
.relocatable_package_version
```

I'm still thinking this is the good place or not, since it still extracts a file on out side of the sub-directory (and even it's dot-file).

To have maximum compatibility with existing releng code, placing the file on top directory should be best.
But we may still can consider to change it, since all existing codes are for non-unified packages (as far as I know).

Or, maybe we can just delete it in unified/install.sh.

Reply:
Oh, I didn't notice the extra ".relocatable_package_version" at the end of the tar command below - github split the line and confused me...
I think that it's not a good idea of putting the file outside the directory you're opening. The whole idea of opening things in a subdirectory is to make everything easy to delete together. It also allows you to open two packages of two versions side by side.
So I think the right thing to do is to put it in the subdirectory with everything else, but I have to admit I don't know why this file exists in the first place or how it is used, so maybe I'm missing something.

Reply:
> I don't know why this file exists in the first place or how it is used, so maybe I'm missing something

We introduced  .relocatable_package_version when we modify relocatable package (which means change directory strucutre, introduce new parameter in install.sh, etc).
We use this to handle different version package in single test code, like this:
```
       package_version_file = "{}/.relocatable_package_version".format(target_dir)
        if os.path.exists(package_version_file):
            with open(package_version_file) as f:
                package_version = packaging.version.parse(f.read().strip())
            if package_version > packaging.version.parse('3'):
                print(f'Unknown relocatable package format version: {package_version}')
                sys.exit(1)
            elif package_version >= packaging.version.parse('2.2'):
                some_test()
            elif package_version >= packaging.version.parse('2.1'):
                other_test()
            ....
```

At least for now the file is only used on tests for non-unified packages (scylla, jmx, tools and python3).
On unified package, it also contains the file  but no tests refers it.
So that's why we may have a change to move the file to sub-directory.

Anyway, I will send a fix to change the file destination to the sub-directory.